template/e2e-tests/tests/landingPageTests.spec.ts

```diff
@@ -57,17 +57,25 @@ test.describe('cookie consent tests', () => {
     // Check that the Cookie Consent cookie is set. This should happen immediately, and then the GA cookies will get set after it, dynamically.
     expect(cookieObject.categories.includes('analytics')).toBeTruthy();
 
-    // Wait for Google Analytics cookies to be set after accepting
-    const MAX_TIME_MS = 45000;
+    const areGaCookiesSet = (cookies: Cookie[]) => {
+      const gaCookiesArr = cookies.filter((c) => c.name.startsWith('_ga'));
+      return gaCookiesArr.length === 2; // GA cookies are _ga and _ga_<GA_ANALYTICS_ID>
+    };
+
     const startTime = Date.now();
-    while (cookies.length < 3) { // 3 cookies will be set: cc_cookie, _ga, _ga_xxx
-      if (Date.now() - startTime > MAX_TIME_MS) {
-        throw new Error('Timeout: Google Analytics cookies not set.');
-      }
-      await new Promise((resolve) => setTimeout(resolve, 1000));
+    const MAX_TIME_MS = 45000;
+    let timeElapsed = 0;
+
+    while (!areGaCookiesSet(cookies) && timeElapsed < MAX_TIME_MS) {
+      await new Promise((resolve) => setTimeout(resolve, 1000)); // wait for 1 second before checking again
+      timeElapsed = Date.now() - startTime;
       cookies = await context.cookies();
     }
-    let gaCookiesArr = cookies.filter((c) => c.name.startsWith('_ga'));
-    expect(gaCookiesArr.length).toBe(2);
+
+    if (timeElapsed >= MAX_TIME_MS) {
+      throw new Error('Timeout: Google Analytics cookies not set.');
+    }
+
+    expect(areGaCookiesSet(cookies)).toBeTruthy();
```

Comment:
The way you have it, this expect will always be truth, so it is not really doing anything, except for telling playwright that test passed.

You could do this though:

```suggestion
    expect(timeElapsed).toBeLessThan(MAX_TIME_MS);
```